bin/all_sky_search/pycbc_fit_sngls_over_multiparam

```diff
@@ -284,13 +314,66 @@ if len(parvals) == 1 and args.smoothing_method == 'smooth_tophat':
     invsum = invalphan.cumsum()
     num = right - left
 
+    logging.info("Smoothing ...")
     ntotal_smoothed = (ntsum[right] - ntsum[left]) / num
     nabove_smoothed = (nasum[right] - nasum[left]) / num
     invmean = (invsum[right] - invsum[left]) / num
     alpha_smoothed = nabove_smoothed / invmean
 
+elif numpy.isfinite(_smooth_cut[args.smoothing_method]):
+    c = _smooth_cut[args.smoothing_method]
+    cut_lengths = [s * c for s in args.smoothing_width]
+    # Find the "longest" dimension in cut lengths
+    sort_dim = numpy.argmax([(v.max() - v.min()) / c
+                              for v, c in zip(parvals, cut_lengths)])
+    logging.info("Sorting / Cutting on dimension %s", parnames[sort_dim])
+
+    # Sort parvals by the sort dimension
+    par_sort = numpy.argsort(parvals[sort_dim])
+    parvals = [p[par_sort] for p in parvals]
+
+    # We will want this later to undo the sorting
+    unsort = numpy.argsort(par_sort)
+    # For each template, find the range of nearby templates which fall within
+    # the chosen window.
+    lefts = numpy.searchsorted(parvals[sort_dim],
+            parvals[sort_dim] - cut_lengths[sort_dim])
+    rights = numpy.searchsorted(parvals[sort_dim],
+            parvals[sort_dim] + cut_lengths[sort_dim])
+    n_removed = len(parvals[0]) - rights + lefts
+    logging.info("Cutting between %d and %d templates for each smoothing",
+                 n_removed.min(), n_removed.max())
+    # Sort the values to be smoothed into paramaeter's order
+    nabove = nabove[par_sort]
+    invalphan = invalphan[par_sort]
+    ntotal = ntotal[par_sort]
```

Comment:
code is easier to read if the quantities are generally in the same order - I would do ntotal, nabove, invalpha

Reply:
or change the code from 'Precompute the sums ..' onward to use the order they are given to the functions, ie nabove, invalpha, ntotal